R/target_market_share.R

```diff
@@ -401,3 +407,49 @@ aggregate_by_loan_id <- function(data) {
     ) %>%
     ungroup()
 }
+
+check_unexpected_columns <- function(data) {
+
+  possible_matched_columns <- c(
+    "id_loan",
+    "id_direct_loantaker",
+    "name_direct_loantaker",
+    "id_intermediate_parent_1",
+    "name_intermediate_parent_1",
+    "id_ultimate_parent",
+    "name_ultimate_parent",
+    "loan_size_outstanding",
+    "loan_size_outstanding_currency",
+    "loan_size_credit_limit",
+    "loan_size_credit_limit_currency",
+    "sector_classification_system",
+    "sector_classification_input_type",
+    "sector_classification_direct_loantaker",
+    "fi_type",
+    "flag_project_finance_loan",
+    "name_project",
+    "lei_direct_loantaker",
+    "isin_direct_loantaker",
+    "id_2dii",
+    "level",
+    "sector",
+    "sector_ald",
+    "name",
+    "name_ald",
+    "score",
+    "source",
+    "borderline"
+    )
+
+unexpected_names <- setdiff(names(data), possible_matched_columns)
+
+  if (length(unexpected_names) != 0) {
+    abort(
+      glue("Loanbook has unexpected names: `{unexpected_names}`."),
```

Comment:
I haven't yet fully understood how an unexpected column caused the problem -- I plan to explore a bit. But this solution is likely to break existing user's code, so if we go this route we may need to help users a bit more. Maybe add a hint to show the code a user would run to remove invalid columns?

Reply:
I think the answer to this question actually lies in the open PR #270
I made a hasty solution, where I aggregated by grouping by all but a few columns (using `group_by_at()` and `setdiff()`) without thinking about the downstream implications of such a decision. In hindsight, this was actually a pretty huge mistake IMO.